### PR TITLE
Debian python3 fixes

### PR DESCRIPTION
**Pull request by m-bucher**
Some fixes I stumbled over while trying to build `subscription-manager` for Ubuntu 20 with pyhton3

### CODE REVIEW

**Comment by jirihnidek** [comment]
@candlepin-bot ok to test

File: Makefile [hunk 1]
```diff
@@ -24,7 +24,11 @@ OS_DIST ?= $(shell rpm --eval='%dist')
 
 PYTHON_VER ?= $(shell $(PYTHON) -c 'import sys; print("python%s.%s" % sys.version_info[:2])')
 
-PYTHON_SITELIB ?= $(PREFIX)/lib64/$(PYTHON_VER)/site-packages
+ifeq ($(filter-out debian ubuntu,$(OS)),)
+  PYTHON_SITELIB ?= $(PREFIX)/lib/$(PYTHON_VER)/site-packages
+else
+  PYTHON_SITELIB ?= $(PREFIX)/lib64/$(PYTHON_VER)/site-packages
+endif
```
Comment (jirihnidek) [inline Makefile]: This looks ok.

File: Makefile [hunk 2]
```diff
@@ -117,8 +121,8 @@ build-subpackages:
 	for subpackage in $(SUBPACKAGES); \
 	do \
 	    pushd $$subpackage; \
-	    $(PYTHON) ./setup.py clean --all \
-	    $(PYTHON) ./setup.py build --root=$(DESTDIR) --prefix=$(PREFIX); \
+	    $(PYTHON) ./setup.py clean --all; \
+	    $(PYTHON) ./setup.py build; \
```
Comment (jirihnidek) [inline Makefile]: Why did you remove `--root` and `--prefix` arguments? Did it cause any issue?

File: debian-stuff/katello [hunk 3]
```diff
@@ -41,7 +40,7 @@ class pkg_acquire_method:
     __eof = False
 
     def __init__(self):
-        print "100 Capabilities\nVersion: 1.0\nSingle-Instance: true\n\n",
+        sys.stdout.write("100 Capabilities\nVersion: 1.0\nSingle-Instance: true\n\n")
```
Comment (jirihnidek) [inline debian-stuff/katello]: I would recommend to add following line to the first line of this file:

```python
from __future__ import print_function
```
Then you can use print() method in Python 2 and Python 3 too.

File: Makefile (hunk 2 above)
Comment (m-bucher) [reply Makefile]: I think they were not listed as parameters in my case, but I might have to reevaluate that.

**Comment by jirihnidek** [comment]
@m-bucher Can you please also add instruction to e.g. REDME.Debian (or extend README.md) how to build subscription-manager on Debian (what packages are required and how to install them)?

**Comment by m-bucher** [comment]
> @m-bucher Can you please also add instruction to e.g. REDME.Debian (or extend README.md) how to build subscription-manager on Debian (what packages are required and how to install them)?

That is actually on my ToDo list.

However, the debian-packaging stuff also needs some work to function properly.

I will see if I can manage to work on that, but yes at least a little bit of documentation might be a good start :smile:

File: debian-stuff/katello (hunk 3 above)
Comment (m-bucher) [reply debian-stuff/katello]: Done, thanks

File: Makefile (hunk 2 above)
Comment (m-bucher) [reply Makefile]: To be fair, this was not the only thing I had to adapt for Ubuntu 20.04 (e.g. using `python3` instead of `python`), but this was the output I got:
```
for subpackage in syspurpose; \
do \
    pushd $subpackage; \
    python3 ./setup.py clean --all; \
    python3 ./setup.py build  --root=/ --prefix=/usr; \
    popd; \
done;
~/subscription-manager/syspurpose ~/subscription-manager
running clean
removing 'build/lib' (and everything under it)
'build/bdist.linux-x86_64' does not exist -- can't clean it
'build/scripts-3.8' does not exist -- can't clean it
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: option --root not recognized
~/subscription-manager
```

Comment (jirihnidek) [reply Makefile]: OK, I see. Can you please make this change platform specific? We try to do some tests with virtual environments (#2228) and this change could break it.

**Comment by jirihnidek** [comment]
@m-bucher Please fix the issue mentioned in https://github.com/candlepin/subscription-manager/pull/2226#discussion_r397112671 and please squash commits. It can go to master after these requests are addressed.

File: Makefile (hunk 2 above)
Comment (m-bucher) [reply Makefile]: Came back to me now: `--root` and `--prefix` are valid parameters for the `install` subcommand.
So it should be save to remove them when using only the `build`-subcommand.

The `install`, which needs those options, is done in the `install-subpackages-via-setup`-stage:
https://github.com/candlepin/subscription-manager/blob/70d7c73d3a2bceef661c409ce687be3aa92c9b26/Makefile#L343-L349

**Comment by m-bucher** [comment]
> @m-bucher Can you please also add instruction to e.g. `README.Debian` (or extend `README.md`) how to build subscription-manager on Debian (what packages are required and how to install them)?

Added #2231

File: Makefile (hunk 2 above)
Comment (m-bucher) [reply Makefile]: removed parameters only for debian-distributions.

I assume the `;` is save to append at the `setup.py clean` line!?

File: Makefile (hunk 1 above)
Comment (m-bucher) [reply Makefile]: Using `$(OS_DIST)` variable now, because it seems more readable.

Debian distributions will have to set it from outside and since it is already so in the debian-build-instructions this should be save:
https://github.com/candlepin/subscription-manager/blob/32a9916998f99c99ddfc0469b004fd7132cfdf96/contrib/debian/rules#L21-L24

File: Makefile [hunk 4]
```diff
@@ -117,8 +121,12 @@ build-subpackages:
...
 	do \
 	    pushd $$subpackage; \
-	    $(PYTHON) ./setup.py clean --all \
-	    $(PYTHON) ./setup.py build --root=$(DESTDIR) --prefix=$(PREFIX); \
+	    $(PYTHON) ./setup.py clean --all; \
+	    if [ '$(OS_DIST)' = 'debian' ]; then \
+	        $(PYTHON) ./setup.py build; \
+	    else; \
+	        $(PYTHON) ./setup.py build --root=$(DESTDIR) --prefix=$(PREFIX); \
+	    fi; \
```
Comment (jirihnidek) [inline Makefile]: Ah, I'm so sorry that I bothered you with this request. Please change it this to:
```
$(PYTHON) ./setup.py build; \
```
The `--root` and `--prefix` options are not supported for `build` command on any platform.

File: Makefile (hunk 1 above)
Comment (jirihnidek) [reply Makefile]: OK :smiley: :+1:

File: Makefile (hunk 4 above)
Comment (m-bucher) [reply Makefile]: done :smiley:
